vkcs/client.go

```diff
@@ -0,0 +1,22 @@
+package vkcs
+
+import "github.com/gophercloud/gophercloud"
+
+func initClientOpts(client *gophercloud.ProviderClient, eo gophercloud.EndpointOpts, clientType string) (*gophercloud.ServiceClient, error) {
+	sc := new(gophercloud.ServiceClient)
+	eo.ApplyDefaults(clientType)
+	url, err := client.EndpointLocator(eo)
+	if err != nil {
+		return sc, err
+	}
+	sc.ProviderClient = client
+	sc.Endpoint = url
+	sc.Type = clientType
+	return sc, nil
+}
+
+func newPublicDNSV2(client *gophercloud.ProviderClient, eo gophercloud.EndpointOpts) (*gophercloud.ServiceClient, error) {
+	sc, err := initClientOpts(client, eo, "publicdns")
+	sc.ResourceBase = sc.Endpoint + "v2/"
```

Comment:
should not we ensure trailing / in sc.Endpoint?

Reply:
It is done by gophercloud SDK:
https://github.com/gophercloud/gophercloud/blob/72205202bfeae679f113e0354bbca444b74d8449/util.go#L51-L56
```
func NormalizeURL(url string) string {
	if !strings.HasSuffix(url, "/") {
		return url + "/"
	}
	return url
}
```